test/parallel/test-inspector-network-resource.js

```diff
@@ -0,0 +1,178 @@
+    params: { url: 'http://localhost:3000/does-not-exist.js' },
+  });
+  assert.strictEqual(resource.success, false);
+  assert(!resource.stream, 'No stream should be returned for missing resource');
+  await session.send({ method: 'Debugger.resume' });
+  await session.waitForDisconnect();
+});
+
+test('should error or return empty for wrong stream id', async () => {
+  const { session } = await setupSessionAndPauseAtEvalLastLine(script);
+  const { resource } = await session.send({
+    method: 'Network.loadNetworkResource',
+    params: { url: resourceUrl },
+  });
+  assert(resource.success);
+  const bogus = '999999';
+  const result = await session.send({ method: 'IO.read', params: { handle: bogus } });
+  assert(result.eof, 'Should be eof for bogus stream id');
+  assert.strictEqual(result.data, '');
+  await session.send({ method: 'IO.close', params: { handle: resource.stream } });
+  await session.send({ method: 'Debugger.resume' });
+  await session.waitForDisconnect();
+});
+
+test('should support IO.read with size and offset', async () => {
+  const { session } = await setupSessionAndPauseAtEvalLastLine(script);
+  const { resource } = await session.send({
+    method: 'Network.loadNetworkResource',
+    params: { url: resourceUrl },
+  });
+  assert(resource.success);
+  assert(resource.stream);
+  const expected = fs.readFileSync(resourcePath, 'utf8');
+  let result = await session.send({ method: 'IO.read', params: { handle: resource.stream, size: 5 } });
+  assert.strictEqual(result.data, expected.slice(0, 5));
+  result = await session.send({ method: 'IO.read', params: { handle: resource.stream, offset: 5, size: 5 } });
+  assert.strictEqual(result.data, expected.slice(5, 10));
+  result = await session.send({ method: 'IO.read', params: { handle: resource.stream, offset: 10 } });
+  assert.strictEqual(result.data, expected.slice(10));
+  await session.send({ method: 'IO.close', params: { handle: resource.stream } });
+  await session.send({ method: 'Debugger.resume' });
+  await session.waitForDisconnect();
+});
+
+test('should load resource put from another thread', async () => {
+  const workerScript = `
+    console.log('this is worker thread');
+    debugger;
+  `;
+  const script = `
+  const { NetworkResources } = require('node:inspector');
+  const fs = require('fs');
+  NetworkResources.put('${resourceUrl}', fs.readFileSync('${resourcePath.replace(/\\/g, '\\').replace(/'/g, "\\'")}', 'utf8'));
+  const { Worker } = require('worker_threads');
+  const worker = new Worker(\`${workerScript}\`, {eval: true});
+  `;
+  const instance = new NodeInstance([
+    '--experimental-inspector-network-resource',
+    '--experimental-worker-inspection',
+    '--inspect-brk=0',
+  ], script);
+  const session = await instance.connectInspectorSession();
+  await setupInspector(session);
+  await session.waitForNotification('Debugger.paused');
+  await session.send({ method: 'Debugger.resume' });
+  const sessionId = '1';
```

Comment:
I think we should retrieve the `seesionId` from the first `Target.attachedToTarget` event, instead of hard-coding it here.

Reply:
Made change!